F/finufft/build_tarballs.jl

```diff
@@ -70,4 +72,4 @@ dependencies = [
 
 # Build the tarballs, and possibly a `build.jl` as well.
 build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies;
-               preferred_gcc_version=v"8", julia_compat="1.6", augment_platform_block)
+               preferred_gcc_version=v"12", julia_compat="1.6", augment_platform_block)
```

Comment:
GCC 12 with Julia v1.6 is going to be problematic for C++ code: https://github.com/JuliaPackaging/Yggdrasil/blob/c3f5c0ce1e1c2a275a3785022b08dcda1de56a0a/CONTRIBUTING.md#compatibility-tips.  Can we use GCC 10?

Reply:
Should work! Updated now